analysis/bulk-deconvolution/exploratory-notebooks/explore-epic-results.Rmd

```diff
@@ -0,0 +1,401 @@
+  "Monocytes" = "grey30", 
+  "Neutrophils" = "grey60"
+)  
+celltype_levels <- c(names(color_map))
+  
+  
+tref_barplot <- make_epic_celltype_barplot(
+  epic_otherna_df |> dplyr::filter(reference == "TRef"),
+  xvar = sample_id, 
+  yvar = fraction, 
+  facetvar = project_diagnosis,
+  fillvar = forcats::fct_relevel(epic_celltype, celltype_levels),
+  plot_title = "TRef reference", 
+  color_map = color_map
+) 
+  
+bref_barplot <- make_epic_celltype_barplot(
+  epic_otherna_df |> dplyr::filter(reference == "BRef"),
+  xvar = sample_id, 
+  yvar = fraction, 
+  facetvar = project_diagnosis,
+  fillvar = forcats::fct_relevel(epic_celltype, celltype_levels),
+  plot_title = "BRef reference", 
+  color_map = color_map
+) 
+
+tref_barplot / bref_barplot
+```
+
+Generally speaking, `TRef` identifies more cell types compared to `BRef`, although whether it _should_ be classifying all those cells is an open question to bear in mind!
+
+Let's compare the proportions of shared cell types inferred between references a bit closely with scatterplots:
+
+```{r fig.width = 12, fig.height = 14}
+shared_celltypes <- c("Bcells", "CD4_Tcells", "CD8_Tcells", "NKcells", "otherCells")
+
+
+compare_reference_celltype <- function(celltype, epic_df) {
+  epic_other_df <- epic_df |>
+    dplyr::filter(epic_celltype == celltype) |>
+    dplyr::select(project_diagnosis, sample_id, reference, fraction) |>
+    dplyr::group_by(project_diagnosis, sample_id) |>
+    tidyr::pivot_wider(
+      names_from = reference, 
+      values_from = fraction
+    ) |>
+    dplyr::mutate(higher_value = ifelse(BRef > TRef, "BRef", "TRef"))
+  
+  ggplot(epic_other_df) + 
+    aes(x = TRef, y = BRef, color = higher_value) + 
+    geom_point(alpha = 0.5) + 
+    geom_abline() +
+    labs(
+      x = "TRef fraction",
+      y = "BRef fraction", 
+      title = celltype
+    ) +
+    facet_wrap(vars(project_diagnosis), nrow = 1)
+}
+
+shared_celltypes |>
+  purrr::map(compare_reference_celltype, epic_df) |>
+  patchwork::wrap_plots(nrow = length(shared_celltypes), guides = "collect")
+```
+From all above comparisons, we see these trends:
+
+* `Bcell`, `CD4_Tcells`,  and `NKcell` proportions are roughly similar between references and low across all samples
+  * In a handful of CNS tumor samples, `TRef` estimates higher `BCell` proportions
+  * In a handful of CNS tumor samples, `TRef` estimates higher `BCell` proportions
+  * It does look like `CD4_Tcells` estimates are a bit higher for `TRef`, but there are two osteosarcoma samples which show really high values from `BRef`
+* `CD8_Tcells` are mostly similar with some additional trends:
+  * Vaues are generally higher for `BRef` for Wilms Tumor samples
+  * Values are generally higher for `TRef` and both CNS tumor and osteosarcoma samples
+* As stated, `TRef` classifies more cells than does `BRef` for all but one osteosarcoma sample
+
+
+**Based on these comparisons, we'll move forward with the `TRef` reference.**
```

Comment:
I don't disagree that Tref is the better option, but I think that's more because of the cell types that Tref has that Bref doesn't have NOT because Tref identifies a higher proportion of cells for the shared cell types. I think it's probably more important to think about the biology of each of these tumors and what normal cell types we may expect to see.

Reply:
In addition to the shared phenotypes, I would look at the monocyte and macrophage fraction in each sample. My guess is that there's a correlation in samples that have monocytes in Bref and macrophages in Tref. Although the macrophage population is probably consistently smaller, I would expect there to be correlation. This may not be important at this point to dive into, but I think that's an important thing to note.

Reply:
> This may not be important at this point to dive into, but I think that's an important thing to note.

This is an interesting question. Agree, this is not the time for a deep dive, but certainly worth a plot.